File: Bymed.Application/Products/ProductDto.cs
namespace Bymed.Application.Products;

public sealed record ProductDto
{
    public required Guid Id { get; init; }
    public required string Name { get; init; }
    public required string Slug { get; init; }
    public required string Description { get; init; }
    public required Guid CategoryId { get; init; }
    public required string CategoryName { get; init; }
    public decimal Price { get; init; }
    public string Currency { get; init; } = string.Empty;
    public int InventoryCount { get; init; }
    public int LowStockThreshold { get; init; }
    public bool IsAvailable { get; init; }
    public string? Sku { get; init; }
    public IReadOnlyDictionary<string, string>? Specifications { get; init; }
}

File: bymed-admin/src/app/shared/models/product.dto.ts
export interface ProductImageDto {
  readonly id: string;
  readonly productId: string;
  readonly url: string;
  readonly altText: string;
  readonly displayOrder: number;
}

export interface ProductDto {
  readonly id: string;
  readonly name: string;
  readonly slug?: string;
  readonly description?: string;
  readonly sku?: string;
  readonly categoryId: string;
  readonly categoryName?: string;
  readonly primaryImageUrl?: string;
  readonly images?: ProductImageDto[];
  readonly price: number;
  readonly currency: string;
  readonly inventoryCount: number;
  readonly lowStockThreshold: number;
  readonly isAvailable: boolean;
}

export interface CreateProductRequestDto {
  readonly name: string;
  readonly slug: string;
  readonly description: string;
  readonly categoryId: string;
  readonly price: number;
  readonly inventoryCount: number;
  readonly lowStockThreshold: number;
  readonly sku?: string | null;
  readonly currency?: string | null;
  readonly specifications?: Record<string, string> | null;
}

export interface UpdateProductRequestDto {
  readonly name: string;
  readonly slug: string;
  readonly description: string;
  readonly categoryId: string;
  readonly price: number;
  readonly lowStockThreshold: number;
  readonly sku?: string | null;
  readonly specifications?: Record<string, string> | null;
}

export interface BulkDeleteProductsRequestDto {
  readonly productIds: string[];
}

export interface BulkSetProductAvailabilityRequestDto {
  readonly productIds: string[];
  readonly isAvailable: boolean;
}

export interface BulkOperationResultDto {
  readonly requestedCount: number;
  readonly processedCount: number;
  readonly notFoundCount: number;
}

export interface ImportProductsResultDto {
  readonly importedCount: number;
  readonly updatedCount: number;
  readonly failedCount: number;
  readonly errors: string[];
}
